### PR TITLE
Healthcheck

### PR DESCRIPTION
**Pull request by ThorArakis**
@hatchetation @dempe

### CODE REVIEW

**Comment by ThorArakis** [comment]
@hatchetation healthcheck looks like:
```
{ maintenance_mode: false }
```
for now.

**Comment by dempe** [comment]
👍

File: kong/plugins/maintenance/handler.lua [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+local os_date = os.date
+local tostring = tostring
+local ngx_log = ngx.log
+local ngx_timer_at = ngx.timer.at
+local ngx_socket_udp = ngx.socket.udp
+local table_concat = table.concat
+local table_insert = table.insert
+
+local function getHostname()
+  local f = io.popen ("/bin/hostname")
```
Comment (hatchetation) [inline kong/plugins/maintenance/handler.lua]: I feel dense ... where is this hostname being used? Is this the only way to get it? Kong doesn't know its hostname already internally? 

Primary concern here is that if kong is under sufficient memory pressure forking can fail.  (See [Shelling Out Sucks](http://julialang.org/blog/2012/03/shelling-out-sucks) for a nice long exploration of the ways shelling out can fail.)

File: kong/plugins/jwt/api.lua [hunk 2]
```diff
@@ -8,7 +8,7 @@ return {
     end,
 
     GET = function(self, dao_factory)
-      crud.paginated_set(self, dao_factory.jwt_secrets)
+      return crud.paginated_set(self, dao_factory.jwt_secrets)
```
Comment (hatchetation) [inline kong/plugins/jwt/api.lua]: Including this change - intentional?

File: kong/plugins/maintenance/schema.lua [hunk 3]
```diff
@@ -0,0 +1,15 @@
+local ALLOWED_LEVELS = { "debug", "info", "notice", "warning", "err", "crit", "alert", "emerg" }
+
+return {
+  fields = {
+    host = { type = "string", default = "logs-01.loggly.com" },
```
Comment (hatchetation) [inline kong/plugins/maintenance/schema.lua]: Schema need updating?

File: kong/plugins/jwt/api.lua (hunk 2 above)
Comment (ThorArakis) [inline kong/plugins/jwt/api.lua]: Unintentional, thanks for catching. fixing!

File: kong/plugins/maintenance/handler.lua (hunk 1 above)
Comment (ThorArakis) [inline kong/plugins/maintenance/handler.lua]: This is actually code we stole from a different kong plugin. The intent was to use it if the cache was somehow getting shared through the cluster. After the discussion with you yesterday we probably won't need to use this. I'll remove from the PR for now and add back in if needed.

File: kong/plugins/maintenance/schema.lua (hunk 3 above)
Comment (ThorArakis) [inline kong/plugins/maintenance/schema.lua]: doh copy + 🍝

**Comment by ThorArakis** [comment]
@hatchetation PR updated

**Comment by dempe** [comment]
👍

**Pull request merged**